Parent/Parent/Dashboard/DashboardViewController.swift

```diff
@@ -256,64 +256,6 @@ class DashboardViewController: UIViewController {
 
         tabsController.viewControllers = [ courses, calendar, alerts ]
     }
-
-    @objc func actionDidTapAddStudent() {
-        if ExperimentalFeature.parentQRCodePairing.isEnabled {
-            let picker = BottomSheetPickerViewController.create()
-            picker.addAction(
-                image: nil,
-                title: NSLocalizedString("QR Code", comment: ""),
-                accessibilityIdentifier: "DashboardViewController.addStudent.qrCode"
-            ) { [weak self] in
-                self?.scanQRCode()
-            }
-            picker.addAction(
-                image: nil,
-                title: NSLocalizedString("Pairing Code", comment: ""),
-                accessibilityIdentifier: "DashboardViewController.addStudent.pairingCode"
-            ) { [weak self] in
-                self?.addStudentController.actionAddStudent()
-            }
-            env.router.show(picker, from: self, options: .modal())
-        } else {
-            addStudentController.actionAddStudent()
-        }
-    }
-
-    func scanQRCode() {
-        let scanner = ScannerViewController()
-        scanner.delegate = self
-        self.env.router.show(scanner, from: self, options: .modal(.fullScreen))
-    }
-}
-
-extension DashboardViewController: ScannerDelegate, ErrorViewController {
-    func scanner(_ scanner: ScannerViewController, didScanCode code: String) {
-        env.router.dismiss(scanner) {
-            guard
-                let components = URLComponents(string: code),
-                let host = components.host,
-                let pairingCode = components.queryItems?.first(where: { $0.name == "code" })?.value
-            else {
-                let error = NSError.instructureError(NSLocalizedString("Could not parse QR code, QR code invalid", comment: ""))
-                self.showError(error)
-                return
-            }
-
-            guard host == self.env.currentSession?.baseURL.host else {
-                let title = NSLocalizedString("Domain mismatch", comment: "")
-                let msg = NSLocalizedString(
-                    """
-                    The student you are trying to add is at a different Canvas institution.
-                    Sign in or create an account with that institution to add this student.
-                    """,
-                    comment: "")
-                self.showAlert(title: title, message: msg)
-                return
-            }
-            self.addStudentController.addPairingCode(code: pairingCode)
-        }
-    }
```

Comment:
Moved all this stuff to AddStudentController so that it can be shared with Manage Students.